sdk/samples/core/mosquitto_multiple_rpc_sample.c

```diff
@@ -0,0 +1,629 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file
+ * @brief RPC Server & Client sample for Mosquitto MQTT that supports multiple commands under the
+ * same subscription
+ *
+ */
+
+#include <az_log_listener.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <uuid/uuid.h>
+
+#include "rpc_client_pending_commands.h"
+#include <azure/az_core.h>
+#include <azure/core/az_log.h>
+#include <azure/core/az_mqtt5_rpc.h>
+#include <azure/core/az_mqtt5_rpc_client.h>
+
+#ifdef _WIN32
+// Required for Sleep(DWORD)
+#include <Windows.h>
+#else
+// Required for sleep(unsigned int)
+#include <unistd.h>
+#endif
+
+// User-defined parameters
+static const az_span client_id = AZ_SPAN_LITERAL_FROM_STR("application-id");
+static const az_span username = AZ_SPAN_LITERAL_FROM_STR("application-id");
+static const az_span hostname = AZ_SPAN_LITERAL_FROM_STR("localhost");
+static const az_span server_client_id = AZ_SPAN_LITERAL_FROM_STR("controller-id");
+static const az_span content_type = AZ_SPAN_LITERAL_FROM_STR("application/json");
+static const az_span start_module_command_name = AZ_SPAN_LITERAL_FROM_STR("startModule");
+static const az_span stop_module_command_name = AZ_SPAN_LITERAL_FROM_STR("stopModule");
+static const az_span server_subscription_topic_format
+    = AZ_SPAN_LITERAL_FROM_STR("device/{executorId}/command/{name}\0");
+static const az_span client_subscription_topic_format
+    = AZ_SPAN_LITERAL_FROM_STR("device/{executorId}/command/{name}/__for_{invokerId}\0");
+static const az_span client_request_topic_format
+    = AZ_SPAN_LITERAL_FROM_STR("device/{executorId}/command/{name}\0");
+
+// Static memory allocation
+static char client_response_topic_buffer[256];
+static char client_request_topic_buffer[256];
+static char client_subscription_topic_buffer[256];
+static uint8_t client_correlation_id_buffers[RPC_CLIENT_MAX_PENDING_COMMANDS]
+                                            [AZ_MQTT5_RPC_CORRELATION_ID_LENGTH];
+
+static char server_subscription_topic_buffer[256];
+// static char response_payload_buffer[256];
+
+// for pending_server_command
+static char server_correlation_id_buffer[256];
+static char server_response_topic_buffer[256];
+static char server_request_topic_buffer[256];
+static char server_request_payload_buffer[256];
+static char server_content_type_buffer[256];
+
+// State variables
+static az_mqtt5_connection mqtt_connection;
+static az_context connection_context;
+
+static az_mqtt5_rpc_server rpc_server;
+static az_mqtt5_rpc_server_policy rpc_server_policy;
+static az_mqtt5_rpc_client rpc_client;
+static az_mqtt5_rpc_client_policy rpc_client_policy;
+
+volatile bool sample_finished = false;
+
+static pending_commands_array client_pending_commands;
+static az_mqtt5_rpc_server_execution_req_event_data pending_server_command;
+
+#ifdef _WIN32
+static timer_t timer; // placeholder
```

Comment:
why is timer in both defs?

Reply:
Resolved windows concerns by using platform implementations